auth/scoperesolver.js

```diff
@@ -252,11 +252,12 @@ class ScopeResolver extends events.EventEmitter {
   }
 
   /**
-   * Return set of scopes that `scopes` can be expanded to when assuming all
-   * authorized roles.
+   * Return a normalized set of scopes that `scopes` can be expanded to when
+   * assuming all authorized roles.
    */
+
   resolve(scopes) {
-    let granted = dfa.sortScopesForMerge(_.clone(scopes));
+    let granted = dfa.sortScopesForMerge(ScopeResolver.normalizeScopes(scopes));
```

Comment:
This is hot code... if we want to do this at the very least leave a `//TODO: Avoid using ScopeResolver.normalizeScopes and normalize scopes using sorted result`

Basically, `ScopeResolver.normalizeScopes` is O(n^2) but after the sort by `dfa.sortScopesForMerge` normalization is trivially done in O(n), see:
https://github.com/taskcluster/taskcluster-auth/blob/3c3b0c9f9a98baa23839ee0d8b0accff878f462f/auth/dfa.js#L152-L166

Technically, this could be done with `mergeScopeSets(dfa.sortScopesForMerge(_.clone(scopes)), [])` which probably isn't even that bad an idea...